L1Trigger/L1TMuonEndCap/python/simEmtfDigis_cfi.py

```diff
@@ -1,47 +1,60 @@
 import FWCore.ParameterSet.Config as cms
 
 # EMTF emulator configuration
-# Three options for CSCInput
-#   * 'simCscTriggerPrimitiveDigis','MPCSORTED' : simulated trigger primitives (LCTs) from re-emulating CSC digis
-#   * 'csctfDigis' : real trigger primitives as received by CSCTF (legacy trigger)
-#   * 'emtfStage2Digis' : real trigger primitives as received by EMTF, unpacked in EventFilter/L1TRawToDigi/
 
 # Check that proper switches are implemented in L1Trigger/Configuration/python/customiseReEmul.py - AWB 02.06.17
 
+###############################################################################################
+###  IMPORTANT!!! Any changes to this file should be reflected in the 2016, 2017, and 2018  ###
+###               configurations in configure_by_fw_version in src/SectorProcessor.cc       ###
+###############################################################################################
+
 simEmtfDigisMC = cms.EDProducer("L1TMuonEndCapTrackProducer",
     # Verbosity level
     verbosity = cms.untracked.int32(0),
 
+    # Configure by firmware version, which may be different than the default parameters in this file
+    FWConfig = cms.bool(True),
+
     # Input collections
+    # Three options for CSCInput
+    #   * 'simCscTriggerPrimitiveDigis','MPCSORTED' : simulated trigger primitives (LCTs) from re-emulating CSC digis
+    #   * 'csctfDigis' : real trigger primitives as received by CSCTF (legacy trigger), available only in 2016 data
+    #   * 'emtfStage2Digis' : real trigger primitives as received by EMTF, unpacked in EventFilter/L1TRawToDigi/
     CSCInput = cms.InputTag('simCscTriggerPrimitiveDigis','MPCSORTED'),
     RPCInput = cms.InputTag('simMuonRPCDigis'),
     GEMInput = cms.InputTag('simMuonGEMPadDigis'),
 
     # Run with CSC, RPC, GEM
-    CSCEnable = cms.bool(True),
-    RPCEnable = cms.bool(True),
-    GEMEnable = cms.bool(False),
+    # Not configured using configure_by_fw_version in SectorProcessor.cc, so what happens for 2016 re-emulation? - AWB 26.04.18
+    CSCEnable = cms.bool(True),  # Use CSC LCTs from the MPCs in track-building
+    RPCEnable = cms.bool(True),  # Use clustered RPC hits from CPPF in track-building
+    GEMEnable = cms.bool(False), # Use hits from GEMs in track-building
+
+    # Era (options: 'Run2_2016', 'Run2_2017', 'Run2_2018')
+    Era = cms.string('Run2_2018'),
```

Comment:
@jiafulow these are both questions for you, I think.  The first one is a question from me.
@thomreis how can we access the "global era" inside the EMTF emulator code?  Do you have an example?

Reply:
Hi @abrinke1 no I don't have an example for this, sorry.

Reply:
Hi @abrinke1 no I don't have an example for this, sorry. I just wanted to know if this has been considered.

Reply:
Hi @abrinke1, @thomreis,

Regarding the first question, originally I intended to have all the configurations done in python. But the amount of configurations quickly became too much, and now part of them became hardcoded on the C++ side. If it is really important, we can send "useCSC_", "useRPC_", "useGEM_" parameters to SectorProcessor, and use them here: https://github.com/cms-sw/cmssw/blob/master/L1Trigger/L1TMuonEndCap/src/SectorProcessor.cc#L405-L407

Otherwise, we just need to remember to set it manually from the python side. Remember, 'Era' also has to be set manually.

Regarding the second question, yes, that was the intention. I made sure the 'Era' parameter values match what is used by the global era. But it has never been used that way. There's very little documentation about how the global era works, so I couldn't figure out how to set it automatically based on the global era. But if you have any suggestions, I'd be happy to do it.

Best regards,
Jia Fu

Reply:
@thomreis @jiafulow 
I went ahead and made the change for RPCs, so we don't forget about it.  One could do the same for CSC and GEM, but this is hypothetical, since we certainly never plan to run without CSCs, and GEMs don't come until 2020.
Cheers, Andrew

Reply:
@abrinke1 @thomreis sorry, but I am missing the purpose of this Era parameter, as the era_ variable in the SectorProcessor class does not seem really used.

Reply:
Sorry, this is a parameter which only actually gets used in the "Phase 2" PR which comes next ... I guess the division between PRs was not as clean as it could have been.

Reply:
it is hard to judge without seeing the actual use of the parameter, I just wonder whether you are not moving in C++ what is done with external eras and modifiers. We need to see its use later